php/WP_CLI/Runner.php

```diff
@@ -1778,6 +1778,10 @@ public function action_setup_theme_wp_cli_skip_themes() {
 		foreach ( $hooks as $hook ) {
 			add_filter( $hook, $wp_cli_filter_active_theme, 999 );
 		}
+		// Remove block pattern loading
+		remove_action( 'init', '_register_theme_block_patterns' );
+		remove_action( 'init', 'gutenberg_register_theme_block_patterns' );
+
```

Comment:
Call me a pessimist, but I'm pretty certain that we need more structure and documentation here, as this will not be the last such change given the FSE direction.

```suggestion

		// Remove theme-related actions not directly tied into the theme lifecycle.
		$theme_related_actions = [
			'init' => '_register_theme_block_patterns',          // Block patterns registration in WP Core.
			'init' => 'gutenberg_register_theme_block_patterns', // Block patterns registration in the GB plugin.
		];
		foreach ( $theme_related_actions as $action => $callback ) {
			remove_action( $action, $callback );
		}

```